### PR TITLE
Pointer line length bugs

### PR DESCRIPTION
**Pull request by keveleigh**
Overview
---
Line 176 in LinePointer was adding 0 (the distance between a point and itself) for the raystep that hit the object. This now properly adds the distance between the start of the raystep and the hit point.

Additionally, `GetNormalizedLengthFromWorldLength` was consistently overshooting and providing a slightly larger value than the true normalized length. This code now subtracts the amount that was overshot.

Changes
---
- Related to #2960

### CODE REVIEW

File: Assets/MixedRealityToolkit/_Core/Utilities/Lines/DataProviders/BaseMixedRealityLineDataProvider.cs [hunk 1]
```diff
@@ -300,20 +300,23 @@ public float GetNormalizedLengthFromWorldLength(float worldLength, int searchRes
             Vector3 lastPoint = GetUnClampedPoint(0f);
             float normalizedLength = 0f;
             float distanceSoFar = 0f;
+            float normalizedSegmentLength = 1f / searchResolution;
```
[the remark this answers is not in the document]
Comment (keveleigh) [reply Assets/MixedRealityToolkit/_Core/Utilities/Lines/DataProviders/BaseMixedRealityLineDataProvider.cs]: Added an early return for `searchResolution < 1`

[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Utilities/Lines/DataProviders/BaseMixedRealityLineDataProvider.cs]: Problem is if it tried to divide by 0

[the remark this answers is not in the document]
Comment (keveleigh) [reply Assets/MixedRealityToolkit/_Core/Utilities/Lines/DataProviders/BaseMixedRealityLineDataProvider.cs]: Yes, that's why I added an early return before it tries to divide.

image

[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Utilities/Lines/DataProviders/BaseMixedRealityLineDataProvider.cs]: Ohhh I see.

[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Utilities/Lines/DataProviders/BaseMixedRealityLineDataProvider.cs]: I couldn't tell the relationship of the line changes at first.

**Pull request merged**